### DISCUSSION TITLE
OAuth2 application

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#261] [General]
Hi, @wu-clan 

I've been studying the OAuth2 authentication process recently, hoping to provide identity verification and authorization similar to what Facebook, Google, and GitHub offer. I was wondering if you'd be interested in researching this together? The official tutorials only provide the most basic demos and then stop abruptly, making it challenging to apply them in practical work scenarios.

[https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/#about-third-party-integrations](https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/#about-third-party-integrations)

I found some related work:
[https://docs.pysnippet.org/fastapi-oauth2/](https://docs.pysnippet.org/fastapi-oauth2/)

It's based on social-core implementation, and to achieve our own authorization methods, we'll need to build the backend logic ourselves.
[https://python-social-auth.readthedocs.io/en/latest/backends/index.html#supported-backends](https://python-social-auth.readthedocs.io/en/latest/backends/index.html#supported-backends)

**Comment by wu-clan** [comment]
I'll take a look at this in my spare time, it shouldn't be difficult, but please remember to protect your keys!

**Reply by downdawn** [reply]
Sure, thank you. If you're interested,

I'll start by attempting the commonly used WeChat and QQ logins. My ultimate goal is to have our own SDK that we can offer to third parties for OAuth2 authorization.

**Comment by wu-clan** [comment]
marking：https://github.com/frankie567/httpx-oauth

**Reply by downdawn** [reply]
marking：[https://tomasvotava.github.io/fastapi-sso/how-to-guides/use-with-fastapi-security/](https://tomasvotava.github.io/fastapi-sso/how-to-guides/use-with-fastapi-security/)

**Comment by wu-clan** [comment]
Follow our integration here：https://github.com/fastapi-practices/fastapi_oauth20

**Reply by wu-clan** [reply]
You can learn how to apply for OAuth2 applications here：https://www.justauth.cn/guide/

**Comment by wu-clan** [comment]
Sorry for being late, as the release of formal OAuth2 applications on most platforms requires strict review, and most programs even have to be reviewed for testing, these things have seriously slowed down my progress. I may end up implementing only one or more of them as examples, which will be completed as soon as possible.
